contracts/src/v0.8/workflow/test/WorkflowRegistryManager/WorkflowRegistryManager.getVersionNumberByContractAddressAndChainID.t.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {WorkflowRegistryManager} from "../../dev/WorkflowRegistryManager.sol";
+import {WorkflowRegistryManagerSetup} from "./WorkflowRegistryManagerSetup.t.sol";
+
+contract WorkflowRegistryManager_getVersionNumberByContractAddressAndChainID is WorkflowRegistryManagerSetup {
+  function test_WhenTheContractAddressIsInvalid() external {
+    // It should revert with InvalidContractAddress
+    _deployMockRegistryAndAddVersion(true);
+    vm.expectRevert(abi.encodeWithSelector(WorkflowRegistryManager.InvalidContractAddress.selector, address(0)));
+    s_registryManager.getVersionNumberByContractAddressAndChainID(address(0), s_chainID);
+  }
+
+  // whenTheContractAddressIsValid
+  function test_WhenNoVersionIsRegisteredForTheContractAddressAndChainIDCombination() external {
+    // It should revert with NoVersionsRegistered.
+    _deployMockRegistryAndAddVersion(true);
+    vm.expectRevert(WorkflowRegistryManager.NoVersionsRegistered.selector);
+    s_registryManager.getVersionNumberByContractAddressAndChainID(address(s_mockWorkflowRegistryContract), 20);
+  }
+
+  // whenTheContractAddressIsValid
+  function test_WhenAVersionIsRegisteredForTheContractAddressAndChainIDCombination() external {
+    // It should return the correct version number.
+    _deployMockRegistryAndAddVersion(true);
+    uint32 versionNumber =
+      s_registryManager.getVersionNumberByContractAddressAndChainID(address(s_mockWorkflowRegistryContract), s_chainID);
+    assertEq(versionNumber, 1);
```

Comment:
What happens if I want to register a new version for the same WR contract on the same chain? Does that mean that `getVersionNumberByContractAddressAndChainID` will return the new version number after I register it (maybe adding a test for that)? And will my previous version forever be lost or can I still see it on the contract, e.g. via `getAllVersions`?

Reply:
Hm but if the contract address is the same and the chain id is the same then it can just activate the old one? I guess we don't have validation on uniqueness I think of (chain id + contract address) but I feel like on the product level that makes sense.

Reply:
will punt on making the changes to make it unique but right now since the key is unique there can only be one anyway so it will just override the version with the new one